src/components/pages/Provisioning.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { useEffect } from 'react';
+import MoonLoader from 'react-spinners/MoonLoader';
+import SparkLogo from '../SparkLogo';
+import useStatus from '../../hooks/useStatus';
+import useCluster from '../../hooks/useCluster';
+import useJupyterLabApp from '../../hooks/useJupyterLabApp';
+import { UIStore } from '../../store/UIStore';
+
+const Provisioning: React.FC = () => {
+  const { mutate } = useStatus();
+
+  const poll = () => {
+    mutate().then(val => {
+      // If the status is stopped in the middle of connecting, flag as failed
+      const isWaitingForConnectResponse = UIStore.getRawState().isConnecting;
+      if (!isWaitingForConnectResponse && val?.status === 'STOPPED') {
+        UIStore.update(s => {
+          s.isConnectionFailed = true;
+        });
+      }
+    });
+  };
+
+  useEffect(() => {
```

Comment:
Can you explain what we are doing here (with a comment)?
Also, why are we polling? Can't we have something more efficient (and quick to notice state changes)?
Or is this because we get a cached state and then have to retrieve an updated value?

Reply:
> Also, why are we polling?

The Spark Connect server takes time to start, and we need to poll for the status. For SWAN's current SparkConnector, there is a kernel messaging mechanism to pass messages using websocket. For this extension, since the server extension is the one starting the Spark Connect server process, we don't have websocket to communicate status changes to the frontend. Hence, we need to use polling.

Reply:
We don't but we can. We can always establish one. That was going to be my point 😉

Reply:
You mean making a PR to Jupyter?

Reply:
@etejedor But in this extension, the process which starts the Spark Connect server is the server extension, not the kernel extension. In kernel extension, we can establish a comm to the frontend, just like the current Spark Connector. However, in the server extension, there is no comm like that.

Reply:
There's no way a server extension can communicate with the frontend? So e.g. the Dask lab extension frontend just polls the server extension for updates e.g. on the number of worker cores?

Reply:
Not quite, I think the request would block until the Dask service is running https://github.com/dask/dask-labextension/blob/main/src/clusters.tsx#L443

We could do it similarly, I think, but it would require quite a lot of changes.

Reply:
Ok let's see if @diocas has some idea.